### PR TITLE
Add Backlog connector

### PR DESCRIPTION
**Pull request by huongtn**
## Description

Add backlog connector

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

https://github.com/faros-ai/airbyte-connectors/issues/233

### CODE REVIEW

**Comment by huongtn** [comment]
@tovbinm currently each projects have many versions, milestones and each issues(tasks) have many versions, milestones.
So can I map  versions, milestones to tms_Sprint and tms_Release?

**Review by Roma-Kyrnis — CHANGES_REQUESTED** [review]
I'm trying to write to Faros-destination but got an Axios error. Can you test writting to Faros-destination?

Can you test using acceptance-test? I've got an error running test

File: destinations/faros-destination/src/converters/backlog/common.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  SprintState,
+  TaskCategory,
+  TaskStatusCategory,
+  TaskType,
+  VersionMilestone,
+} from './models';
+
+export class BacklogCommon {
+  // Max length for free-form description text fields such as issue body
+  static readonly MAX_DESCRIPTION_LENGTH = 1000;
```
[the remark this answers is not in the document]
Comment (huongtn) [reply destinations/faros-destination/src/converters/backlog/common.ts:15]: Could you please share Agile Accelerator pull request link?

File: sources/backlog-source/test_files/full_configured_catalog.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
```
[the remark this answers is not in the document]
Comment (huongtn) [reply sources/backlog-source/test_files/full_configured_catalog.json:1]: Currently already get from command, what is your side?

File: sources/backlog-source/test/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    Backlog.instance = BacklogInstance;
+  });
+
+  test('spec', async () => {
+    const source = new sut.BacklogSource(logger);
+    await expect(source.spec()).resolves.toStrictEqual(
+      new AirbyteSpec(readTestResourceFile('spec.json'))
+    );
+  });
+  // test('check connection', async () => {
```
Comment (tovbinm) [inline sources/backlog-source/test/index.test.ts:44]: please fix the tests

**Comment by cjwooo** [comment]
@huongtn Please slack me the config.json you used to run the source acceptance tests. Use https://onetimesecret.com/ to secure the secret and send me the url.

**Comment by huongtn** [comment]
> @huongtn Please slack me the config.json you used to run the source acceptance tests. Use https://onetimesecret.com/ to secure the secret and send me the url.

ok sure

**Comment by huongtn** [comment]
Already test e2e

File: destinations/faros-destination/src/converters/backlog/projects.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        project: {uid, source},
+      },
+    });
+    for (const versionMilestone of project.versionMilestones) {
+      res.push({
+        model: 'tms_Sprint',
+        record: {
+          uid: String(versionMilestone.id),
+          name: versionMilestone.name,
+          description: versionMilestone.description,
```
Comment (Roma-Kyrnis) [inline destinations/faros-destination/src/converters/backlog/projects.ts:54]: Use subString to cut description to default length

File: destinations/faros-destination/src/converters/backlog/projects.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          source,
+        },
+      });
+      if (versionMilestone.releaseDueDate) {
+        res.push({
+          model: 'tms_Release',
+          record: {
+            uid: String(versionMilestone.id),
+            name: versionMilestone.name,
+            description: versionMilestone.description,
```
Comment (Roma-Kyrnis) [inline destinations/faros-destination/src/converters/backlog/projects.ts:67]: And here also: Use subString to cut description to default length

**Review by Roma-Kyrnis — CHANGES_REQUESTED** [review]
Update to main branch similar to BuildKite

File: sources/backlog-source/src/backlog.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+          },
+        }
+      : {
+          params: {
+            updatedSince: lastUpdatedAt ? this.formatDate(startTime) : '',
+          },
+        };
+    const res = await this.httpClient.get<Issue[]>('issues', config);
+    for (const item of res.data) {
+      if (!lastUpdatedAt || new Date(item.updated) >= startTime) {
```
[the remark this answers is not in the document]
Comment (huongtn) [reply sources/backlog-source/src/backlog.ts:99]: Because updatedSince only for Date, we need to check both Date and Time

**Review by cjwooo — APPROVED** [review]

**Pull request merged**